File: UBL/CommonAggregateComponents/LanguageType.cs
using System.ComponentModel.DataAnnotations;
using System.Xml.Linq;
using Hilma.UBL.Attributes;
using Hilma.UBL.Serializers;
using Hilma.UBL.UnqualifiedDataTypes;

namespace Hilma.UBL.CommonAggregateComponents
{
  [Contract]
  public class Language
  {
    [Required]
    public IdentifierType ID { get; set; }

    public string Name { get; set; }

    public CodeType LocaleCode { get; set; }

    public XElement Serialize()
    {
      return new XElement(UblNames.Cac + nameof(Language),
        ID.Serialize(nameof(ID)),
        Name.Serialize(nameof(Name)),
        LocaleCode.Serialize(nameof(LocaleCode)));
    }
  }
}

File: Espd/typings/LanguageType.ts



import { IdentifierType } from './IdentifierType'
import { CodeType } from './CodeType'

import assign from 'lodash.assign'

export class LanguageType { 
    public constructor(init?:Partial<LanguageType>) {
        assign(this, init)
    } 
    
    
    id?: IdentifierType
    name?: string
    localeCode?: CodeType
}